plugins/webhooks/index.js

```diff
@@ -310,6 +311,94 @@ function pullRequestEvent(pluginOptions, request, reply, parsed) {
         .catch(err => reply(boom.wrap(err)));
 }
 
+/**
+ * Check the pipeline have triggered job or not
+ * @method  isTrigger
+ * @param   {Pipeline}  pipeline    The pipeline to check
+ * @param   {String}    startFrom   The trigger name
+ * @returns {boolean}               If true the pipeline has triggered job
+ */
+function isTrigger(pipeline, startFrom) {
+    const nextJobs = workflowParser.getNextJobs(pipeline.workflowGraph, {
+        trigger: startFrom
+    });
+
+    return nextJobs.length > 0;
+}
+
+/**
+ * Get all pipelines which has triggered job
+ * @method  triggerPipelines
+ * @param   {pipelineFactory}   pipelineFactory the pipeline factory to get branch list
+ * @param   {Object}            scmConfig       has the token and scmUri to get branches
+ * @param   {String}            branch          the branch which is committed
+ * @returns {Promise}                           Promise that resolves into triggered pipelines
+ */
+function triggerPipelines(pipelineFactory, scmConfig, branch) {
+    return pipelineFactory.scm.getBranchList(scmConfig)
+        .then((branches) => {
+            const scmUris = [];
+            const splitUrl = scmConfig.scmUri.split(':');
+
+            branches.forEach((b) => {
+                if (b.name !== branch) {
+                    splitUrl[2] = b.name;
+                    scmUris.push(splitUrl.join(':'));
+                }
+            });
+
+            return scmUris;
+        })
+        .then(scmUris => pipelineFactory.list({ scmUri: scmUris }))
+        .then(pipelines => pipelines.filter(p => isTrigger(p, `~commit:${branch}`)))
+        .then((pipelines) => {
+            const scmUri = scmConfig.scmUri;
+
+            return pipelineFactory.get({ scmUri }).then((p) => {
+                if (p) {
+                    pipelines.push(p);
+                }
+
+                return pipelines;
+            });
+        });
+}
+
+/**
+ * Create events for each pipeline
+ * @async createEvents
+ * @param {EventFactory}    eventFactory    to create event
+ * @param {Array}           pipelines       the pipelines to start events
+ * @param {String}          parsed          it have a information to create event
+ * @returns {Promise}                       Promise that resolves into events
+ */
+async function createEvents(eventFactory, pipelines, parsed) {
+    const { branch, sha, username, scmContext, changedFiles } = parsed;
+    const events = [];
+
+    for (let i = 0; i < pipelines.length; i += 1) {
```

Comment:
How about using `forEach`? `i` is used only pipelines index.

Reply:
I try to use forEach, but when I try to use await it failed. So I use basic for.
https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop

Reply:
In this page, the answerer said you should use `for … of`. How about it?

Reply:
When I use `for ... of` , I got `no-restricted-syntax` lint error, thus I use basic `for` now. 
> error  iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations  no-restricted-syntax

Reply:
oh..